contracts/libraries/AaveLenderBorrowerLib.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "./aave/DataTypes.sol";
+import "../interfaces/aave/IReserveInterestRateStrategy.sol";
+import "../interfaces/aave/ILendingPool.sol";
+import "../interfaces/aave/IStakedAave.sol";
+import "../interfaces/aave/IPriceOracle.sol";
+import "../interfaces/aave/IAToken.sol";
+import "../interfaces/aave/IVariableDebtToken.sol";
+import "../interfaces/IOptionalERC20.sol";
+
+import "../interfaces/aave/IProtocolDataProvider.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../WadRayMath.sol";
+
+library AaveLenderBorrowerLib {
+    using SafeMath for uint256;
+    using WadRayMath for uint256;
+    struct CalcMaxDebtLocalVars {
+        uint256 availableLiquidity;
+        uint256 totalStableDebt;
+        uint256 totalVariableDebt;
+        uint256 totalDebt;
+        uint256 utilizationRate;
+        uint256 totalLiquidity;
+        uint256 targetUtilizationRate;
+        uint256 maxProtocolDebt;
+    }
+
+    struct IrsVars {
+        uint256 optimalRate;
+        uint256 baseRate;
+        uint256 slope1;
+        uint256 slope2;
+    }
+
+    function protocolDataProvider()
+        public
+        view
+        returns (IProtocolDataProvider)
+    {
+        return
+            IProtocolDataProvider(0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d);
+    }
+
+    function MAX_BPS() internal view returns (uint256) {
```

Comment:
uff you can't define constants in a library? making it pure might reduce contract size.

Reply:
brain fart :D

you can set constants, you cannot set state variables